module/profile/helper.js

```diff
@@ -0,0 +1,109 @@
+const axios = require('axios')
+const entityFind = require('../../generics/services/entity-management')
+const userServiceUrl = process.env.USER_SERVICE_URL
+
+module.exports = class FormsHelper {
+	/**
+	 * @function read
+	 * @description Fetches and processes user profile data, including location details.
+	 * @param {Object} userId - The user details object containing the user's information.
+	 * @returns {Promise<Object>} - A promise that resolves with the processed user profile data or an error object.
+	 * @throws {Error} - Throws an error if the user details cannot be fetched or processed.
+	 **/
+	static read(userId) {
+		return new Promise(async (resolve, reject) => {
+			try {
+				// Define headers for the user service request
+				let headers = {
+					internal_access_token: 'internal_access_token',
+					'Content-Type': 'application/json',
+				}
+				// Fetch user details from user service
+				const userResponse = await axios.get(
+					userServiceUrl + CONSTANTS.endpoints.USER_SERVICE_URL + userId.userInformation.userId,
+					{ headers }
+				)
+				// Check if user details were fetched successfully
+				if (userResponse.status !== 200) {
+					throw new Error(CONSTANTS.common.STATUS_FAILURE)
+				}
+				const userDetails = userResponse.data.result
+
+				const locationIds = await this.extractLocationIds(userDetails.meta)
```

Comment:
@MallanagoudaB if locationIds array is not empty, then only proceed to next step